src/app/services/product.service.ts

```diff
@@ -12,107 +13,83 @@ export class ProductService {
-    localStorage.setItem(
-      this.productCountKey,
-      JSON.stringify(this.productCount)
+    this.localStorageService.setItem(this.cartProductsKey, this.cartProducts);
+    this.localStorageService.setItem(this.productCountKey, this.productCount);
+  }
+
+  public getProducts(): Observable<ProductData[]> {
+    return this.http.get<ProductData[]>(this.apiurl).pipe(
+      map(response => {
+        return response;
+      }),
+      catchError(error => {
+        throw error;
+      })
     );
   }
 
-  getProducts(): Observable<ProductData[]> {
-    return this.http.get<ProductData[]>(this.apiurl);
-  }
-
-  getSelectedProduct(id: string): Observable<ProductData> {
+  public getSelectedProduct(id: string): Observable<ProductData> {
     return this.http.get<ProductData>(`${this.apiurl}/${id}`);
   }
 
-  createProduct(data: {
-    title: string;
-    price: number;
-    description: string;
-    image: string;
-    category: string;
-  }): Observable<ProductData> {
+  public createProduct(
+    data: Pick<ProductData, 'title' | 'price' | 'description' | 'category'>
+  ): Observable<ProductData> {
     return this.http.post<ProductData>(`${this.apiurl}`, {
       ...data,
     });
   }
 
-  setSelectedProduct(product: ProductData): void {
-    this.selectedProduct = product;
-    this.updateLocalStorage();
-  }
-
-  setSelectedProductToCart(product: cartProductData): void {
-    const existingProductIndex = this.cartProducts.findIndex(
-      (p) => p.id === product.id
-    );
-    console.log(existingProductIndex);
-    if (existingProductIndex !== -1) {
-      // Product already exists in the cart, increase count by one
-      const existingProduct = this.cartProducts[existingProductIndex];
-      const updatedProduct = {
-        ...existingProduct,
-        count: (existingProduct.count || 0) + 1,
-      };
-      this.cartProducts[existingProductIndex] = updatedProduct;
+  public addProductToCart(product: CartProductData): void {
+    const existingProduct = this.cartProducts.find(p => p.id === product.id);
+    if (existingProduct) {
+      existingProduct.count = (existingProduct.count || 1) + 1;
     } else {
-      // Product does not exist in the cart, add it
-      const newProduct = { ...product, count: 1 };
-      this.cartProducts.push(newProduct);
-      this.productCount += 1;
+      this.cartProducts.push({ ...product, count: 1 });
     }
-
     this.updateLocalStorage();
   }
 
-  incrementProductCount(productId: string): void {
+  public incrementProductCount(productId: string): void {
     this.updateProductCount(productId, 1);
   }
 
-  decrementProductCount(productId: string): void {
+  public decrementProductCount(productId: string): void {
     this.updateProductCount(productId, -1);
   }
 
   private updateProductCount(productId: string, countChange: number): void {
-    const index = this.cartProducts.findIndex((p) => p.id === productId);
+    const index = this.cartProducts.findIndex(p => p.id === productId);
```

Comment:
```suggestion
    const index = this.cartProducts.findIndex({id} => id === productId);
```

Reply:
the store is being used so this code has been removed

Reply:
the store is being used so this code has been removed